go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -0,0 +1,268 @@
+		}
+		return bsnc.ensurePodForSecondaryNetwork(pod, true)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+// UpdateSecondaryNetworkResourceCommon updates the specified object in the cluster to its version in newObj
+// according to its type and returns the error, if any, yielded during the object update. This function is
+// called for secondary network only.
+// Given an old and a new object; The inRetryCache boolean argument is to indicate if the given resource
+// is in the retryCache or not.
+func (bsnc *BaseSecondaryNetworkController) UpdateSecondaryNetworkResourceCommon(objType reflect.Type, oldObj, newObj interface{}, inRetryCache bool) error {
+	switch objType {
+	case factory.PodType:
+		oldPod := oldObj.(*kapi.Pod)
+		newPod := newObj.(*kapi.Pod)
+
+		return bsnc.ensurePodForSecondaryNetwork(newPod, inRetryCache || util.PodScheduled(oldPod) != util.PodScheduled(newPod))
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+// DeleteResource deletes the object from the cluster according to the delete logic of its resource type.
+// Given an object and optionally a cachedObj; cachedObj is the internal cache entry for this object,
+// used for now for pods.
+// This function is called for secondary network only.
+func (bsnc *BaseSecondaryNetworkController) DeleteSecondaryNetworkResourceCommon(objType reflect.Type, obj, cachedObj interface{}) error {
+	switch objType {
+	case factory.PodType:
+		var portInfoMap map[string]*lpInfo
+		pod := obj.(*kapi.Pod)
+
+		if cachedObj != nil {
+			portInfoMap = cachedObj.(map[string]*lpInfo)
+		}
+		return bsnc.removePodForSecondaryNetwork(pod, portInfoMap)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+// ensurePodForSecondaryNetwork tries to set up secondary network for a pod. It returns nil on success and error
+// on failure; failure indicates the pod set up should be retried later.
+func (bsnc *BaseSecondaryNetworkController) ensurePodForSecondaryNetwork(pod *kapi.Pod, addPort bool) error {
+
+	// Try unscheduled pods later
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	if !util.PodWantsNetwork(pod) && !addPort {
+		return nil
+	}
+
+	// If a node does not have an assigned hostsubnet don't wait for the logical switch to appear
+	switchName, err := bsnc.getExpectedSwitchName(pod)
+	if err != nil {
+		return err
+	}
+
+	on, network, err := util.PodWantsMultiNetwork(pod, bsnc.NetInfo)
+	if err != nil {
+		// configuration error, no need to retry, do not return error
+		klog.Errorf("Error getting network-attachment for pod %s/%s network %s: %v",
+			pod.Namespace, pod.Name, bsnc.GetNetworkName(), err)
+		return nil
+	}
+
+	if !on {
+		// the pod is not attached to this specific network
+		klog.V(5).Infof("Pod %s/%s is not attached on this network controller %s error (%v) ",
+			pod.Namespace, pod.Name, bsnc.GetNetworkName(), err)
```

Comment:
The `err` is nil here
```suggestion
		klog.V(5).Infof("Pod %s/%s is not attached on this network controller %s",
			pod.Namespace, pod.Name, bsnc.GetNetworkName())
```